barman/server.py

```diff
@@ -797,10 +823,64 @@ def check_postgres(self, check_strategy):
                     hint="please set it to a higher level than 'minimal'",
                 )
 
+    def _check_has_monitoring_privileges(
+        self, check_strategy, remote_status, suffix=None
+    ):
+        check_name = "has monitoring privileges"
+        if suffix is not None:
+            check_name = f"{check_name} ({suffix})"
+        check_strategy.init_check(check_name)
+        if remote_status.get("has_monitoring_privileges"):
+            check_strategy.result(self.config.name, True)
+        else:
+            check_strategy.result(
+                self.config.name,
+                False,
+                hint="privileges for PostgreSQL monitoring functions are "
+                "required (see documentation)",
+                check="no access to monitoring functions",
```

Comment:
We need to change the documentation. Currently it only explains about a whole set of privileges which were required by "has_backup_privileges". The docs should be amended to make a separation of the privileges, and explain which ones should be granted because of what reasons / settings.

Reply:
Yes - thanks for raising this, we'll need to make sure permissions get covered in the documentation for this change. I'll add a commit with the docs before making this ready for review.